docs/how-to-guides/azure-arc-onboarding.md

```diff
@@ -0,0 +1,223 @@
+---
```

Comment:
I would change the whole doc to have only `az` commands. I have tested the following:

```bash
RG_NAME="suraj-azure-arc-test"
CLUSTER_NAME="suraj-em-cluster"
SUB_ID=$(az account show --query id --output tsv)

az provider register --namespace Microsoft.Kubernetes
az provider register --namespace Microsoft.KubernetesConfiguration
az provider register --namespace Microsoft.ExtendedLocation

az provider show -n Microsoft.Kubernetes -o table
az provider show -n Microsoft.KubernetesConfiguration -o table
az provider show -n Microsoft.ExtendedLocation -o table

az group create --name "${RG_NAME}" --location EastUS --output table


az ad sp create-for-rbac -n azure-arc-enabled-kubernetes --skip-assignment -o jsonc > /tmp/sp-cred.json
SP_ID=$(az ad sp list -o tsv --query='[0].objectId' --display-name azure-arc-enabled-kubernetes)

# Assign "Kubernetes Cluster - Azure Arc Onboarding" Role by its identifier
az role assignment create --assignee "${SP_ID}" \
    --role "Kubernetes Cluster - Azure Arc Onboarding" \
    --scope /subscriptions/${SUB_ID}/resourcegroups/${RG_NAME}

az role assignment create --assignee "${SP_ID}" \
    --role "Microsoft.Kubernetes connected cluster role" \
    --scope /subscriptions/${SUB_ID}/resourcegroups/${RG_NAME}


# Copy values from /tmp/sp-cred.json

APPLICATION_ID=$(jq -r .appId /tmp/sp-cred.json)
APPLICATION_PASSWORD=$(jq -r .password /tmp/sp-cred.json)
TENANT_ID=$(jq -r .tenant /tmp/sp-cred.json)

echo "component \"azure-arc-onboarding\" {
  application_client_id = \"${APPLICATION_ID}\"
  application_password  = \"${APPLICATION_PASSWORD}\"
  tenant_id             = \"${TENANT_ID}\"
  resource_group        = \"${RG_NAME}\"
  cluster_name          = \"${CLUSTER_NAME}\"
}
"

az extension add --name k8s-configuration

az k8s-configuration create \
  --name cluster-config \
  --cluster-name "${CLUSTER_NAME}" \
  --resource-group ${RG_NAME} \
  --operator-instance-name cluster-config \
  --operator-namespace cluster-config \
  --repository-url https://github.com/ipochi/azure-arc-nginx-demo \
  --scope cluster \
  --cluster-type connectedClusters \
  --operator-params='--git-branch=main'
```

Reply:
In ^^^ the service principal name `azure-arc-enabled-kubernetes` can be parameterised as well!